### PR TITLE
Update self-host / kubernetes hosting mode docs for 1.0.0-rc.1

### PR DESCRIPTION
**Pull request by youngbupark**
Thank you for helping make the Dapr documentation better!

If you are a new contributor, please see the [this contribution guidance](https://docs.dapr.io/contributing/contributing-docs/) which helps keep the Dapr documentation readable, consistent and useful for Dapr users.

Note that you must see that the suggested changes do not break the website [docs.dapr.io](https://docs.dapr.io). See [this overview](https://github.com/dapr/docs/blob/master/README.md#overview) on how to setup a local version of the website and make sure the website is built correctly.

In addition, please fill out the following to help reviewers understand this pull request:

## Description

* Update self-host mode placement doc
* Update Kubernetes hosting mode for 1.0.0-rc.1

## Issue reference

_Please reference the issue this PR will close: #898

### CODE REVIEW

**Comment by yaron2** [comment]
Shouldn't this go to the v1.0-rc branch? the default docs will point to 0.11.3, I want to make sure that users see the upgrade instructions with the latest stable version. These guidelines shouldn't have version specifics inside of them.

**Comment by youngbupark** [comment]
> Shouldn't this go to the v1.0-rc branch? the default docs will point to 0.11.3, I want to make sure that users see the upgrade instructions with the latest stable version. These guidelines shouldn't have version specifics inside of them.

master is not the stable version. master is WIP branch for dapr/dapr master. If someone want to see the doc 0.11.3, they should checkout v0.11.3 tag.

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md [hunk 1]
```diff
@@ -59,10 +59,10 @@ You can find information [here]({{< ref "install-dapr.md#using-helm-advanced" >}
 When deploying Dapr in a production-ready configuration, it's recommended to deploy with a highly available configuration of the control plane:
 
 ```bash
-helm install dapr dapr/dapr --namespace dapr-system --set global.ha.enabled=true
+helm install dapr dapr/dapr --version=1.0.0-rc.x --namespace dapr-system --set global.ha.enabled=true
```
Comment (orizohar) [inline daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md:62]: Should this specific command apply only to rc.x releases?

Comment (yaron2) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md:62]: It shouldn't..

**Comment by yaron2** [comment]
> > Shouldn't this go to the v1.0-rc branch? the default docs will point to 0.11.3, I want to make sure that users see the upgrade instructions with the latest stable version. These guidelines shouldn't have version specifics inside of them.
> 
> master is not for the stable version. master is WIP branch for dapr/dapr master. If someone wants to see the doc 0.11.3, they should checkout v0.11.3 tag.

Yeah but master branch docs so far had no explicit versioning in these guidelines, so I don't think we should be adding them now.

File: daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md (hunk 1 above)
Comment (youngbupark) [reply daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md:62]: @yaron2 @orizohar / I have talked to ori offline. Without chart version option, user will get 0.11.3 dapr, but doc's master repo will have the information for 1.0.0-rc.x, not for 0.11.3.

To reduce the confusion, we may need to ask users to specify the specific version of chart.

I will change 1.0.0-rc.x to `<Dapr runtime version>` to let user specify the specific version of dapr.

**Comment by youngbupark** [comment]
> > > Shouldn't this go to the v1.0-rc branch? the default docs will point to 0.11.3, I want to make sure that users see the upgrade instructions with the latest stable version. These guidelines shouldn't have version specifics inside of them.
> > 
> > 
> > master is not for the stable version. master is WIP branch for dapr/dapr master. If someone wants to see the doc 0.11.3, they should checkout v0.11.3 tag.
> 
> Yeah but master branch docs so far had no explicit versioning in these guidelines, so I don't think we should be adding them now.

I will remove the specific version (1.0.0-rc.1) but will guide user to set the specific version of dapr chart.

**Review by orizohar — APPROVED** [review]

**Pull request merged**